File: packages/sdk-python/examples/block_rm_rf.py
import sys
import re
import polyhook

event = polyhook.read()

if event.tool == "bash" and re.search(
    r"rm\s+-rf\s+/", event.input.get("command", "") if event.input else ""
):
    polyhook.respond(polyhook.block("Refusing to delete from root"))
else:
    polyhook.respond(polyhook.approve())

File: packages/sdk-rust/examples/block_rm_rf.rs
use polyhook::{read, respond, HookResponse};

fn main() {
    let event = match read() {
        Ok(e) => e,
        Err(e) => {
            eprintln!("polyhook: failed to read event: {e}");
            std::process::exit(1);
        }
    };

    let response = match event.tool.as_deref() {
        Some("bash") => {
            let cmd = event
                .input
                .as_ref()
                .and_then(|i| i.get("command"))
                .and_then(|v| v.as_str())
                .unwrap_or("");

            if cmd.contains("rm -rf /") {
                HookResponse::block("Refusing to delete from root")
            } else {
                HookResponse::approve()
            }
        }
        _ => HookResponse::approve(),
    };

    if let Err(e) = respond(&response) {
        eprintln!("polyhook: failed to write response: {e}");
        std::process::exit(1);
    }
}
